### ISSUE TITLE
Override Bootstrap and Use Underlines for Most Hyperlinks

### ISSUE THREAD

**Issue by mtribone** [samvera/hyrax#1255]
### Descriptive summary

In Hyrax 2.0.0 alpha hyperlinks only show the underline on hover and this convention can be problematic for users with differing color vision because the cues use *only* color. Propose using original underlines in addition to color for main content that isn't being displayed using other conventions; for example dropdown menus.

### Rationale

If my color vision is different and I cannot distinguish the "right" color to indicate a clickable link then I do not know intuitively what areas of the content are clickable. If we put back the old school underlines then people with differing color abilities and others will benefit from the visual cues in the interface. See screen shots.

### Expected behavior

I expect to distinguish all of the clickable areas of the interface.

### Actual behavior

I am unsure sometimes of what is clickable so I just click around the interface. For example, the badges indicating open access next to the title or all of the metadata?

### Steps to reproduce the behavior

1. Pick an area of the application like the search results
1. Make it black and white by importing a screen shot into Photoshop or using a browser plugin
1. Point out all of the clickable links in the interface

### Related work

Link to related tickets or prior related work here.

https://github.com/psu-stewardship/scholarsphere/issues/517

https://github.com/psu-stewardship/scholarsphere/commit/ab6b1f91c9d785756719ea2e0b5833a9563671af

**Comment by mtribone** [comment]
![screen shot 2017-06-21 at 2 14 56 pm](https://user-images.githubusercontent.com/4163828/27400033-d2bd1e4e-568c-11e7-88af-017298f62310.png)

**Comment by mtribone** [comment]
![screen shot 2017-06-21 at 2 18 30 pm](https://user-images.githubusercontent.com/4163828/27400052-e0cf0308-568c-11e7-832b-3d5dc749e321.png)

**Comment by mtribone** [comment]
![screen shot 2017-06-21 at 2 19 32 pm](https://user-images.githubusercontent.com/4163828/27400072-f2aa2f1c-568c-11e7-8f5c-1b373ae31566.png)

**Comment by atz** [comment]
@mtribone: I take it your screenshots have injected a logo at each point of complaint?

I get the accessibility concern, but there is a reason bootstrap is the *de facto* baseline for appearance and that underlined links is considered throwback.  It clutters the appearance greatly.  

Also it suggests that when you **do** use underline, that it _always_ signifies links, which is incorrect, in particular for a platform that supports user-supplied content blocks (since the difference in color of the underlined text and the link is not perceptible for exactly the same reasons).  

I am absolutely against underlining button text.  Buttons are already clearly clickable and your tool doesn't have a problem with *some* of them.  Also against additional indication where the "popout" or "new window" icon is already in use as a clear color-independent indicator (e.g. `Related URL`).

As design thing, idk, I want to support accessibility but I think underlines everywhere would destroy the current look.  We're using github right now in an interface that underlines zero links, for example.

**Comment by mtribone** [comment]
@atz Yeah I don't think that everything has to be underlined-perhaps I needed to stress that more. Bootstrap is pretty popular, but that doesn't always make it right.  Hyperlinks being underlined is a pretty basic convention and a pretty useful pattern. We've implemented the change in ScholarSphere if anybody wants to take a peek at what it looks like in the interface.

https://scholarsphere.psu.edu/

**Comment by mtribone** [comment]
As for the icons in the screen shots above, they are indicating contrast errors in the interface. My apologies! I bet that was confusing. The tool can desaturate the page, but only on the contrast error section. Buttons, drop downs, menu bars, etc. already have a convention so I wouldn't underline those things.

**Comment by rjkati** [comment]
In hyrax 3.0.0 rc3, behavior is still the same as initially reported. Hyperlinks only show the underline on hover.

**Comment by jlhardes** [comment]
Decision: No changes to hyperlink styling so this issue is being closed.

The color contrast for hyperlinks in Hyrax passes WCAG 2.0 (foreground color #0453AB contrasts with background color #FFFFFF in a contrast ratio of 7.41:1 according to [WebAIM's contrast checker](https://webaim.org/resources/contrastchecker/). Underlines do appear on hover which seems to be how Bootstrap handles things by default so there would be work to override this styling. Since we have a strong color contrast out of the box, the underline appears on hover, and there would be additional work to override this behavior, we will leave this as-is.
